tests/test_transformers.py

```diff
@@ -516,6 +518,43 @@ def test_xlm(self):
         self.assertLessEqual(test['ppl'], 1.30)
         self.assertGreaterEqual(test['bleu-4'], 0.90)
 
+    def test_asymmetry(self):
```

Comment:
Hi @dexterju :) this PR looks very useful, I really like it! Just a tiny question (reply to me when you have some free time, no worries) why is this function necessary? Is it just a sanity check to verify the parameters are being overridden?

Reply:
Hey, This unit test function will just check these parameters are set correctly. 
We want the logic to be if n-encoder-layers are explicitly given, it should use it, otherwise it should use n-layers instead to ensure backward compatibility (same case for the decoders). It is good to have this test ensuring this logic is respected.

Reply:
Ahhh I see, thanks!